x-pack/plugins/enterprise_search/public/applications/app.test.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { MemoryRouter } from 'react-router-dom';
+
+import { coreMock } from 'src/core/public/mocks';
+import { mountWithIntl } from 'test_utils/enzyme_helpers';
+import { App } from './app';
+
+jest.mock('../applications/app_search', () => ({
+  AppSearch: () => <div>App Search Mock</div>,
+}));
+
+describe('App', () => {
+  let params;
+  let core;
+
+  beforeEach(() => {
+    params = coreMock.createAppMountParamters();
+    core = coreMock.createStart();
+  });
+
+  describe('routing', () => {
+    let route;
+
+    beforeEach(() => {
+      route = null;
+    });
+
+    const subject = () =>
+      mountWithIntl(
+        <MemoryRouter initialEntries={[route]}>
+          <App
+            core={core}
+            params={params}
+            config={{
+              host: 'http://localhost:3002',
+            }}
+          />
+        </MemoryRouter>
+      );
```

Comment:
I'm a little confused on the need for this. 🤔 I'm also maybe slightly worried that we're mocking too much here. Do you think if we move `BrowserRouter` to App (as mentioned in my earlier comment) that we'll negate the need for mounting an alternative router?

Reply:
I moved `BrowserRouter` out so that we could replace it with our own router because:
- I don't know if we'll actually be able to use a BrowserRouter in these tests: https://reacttraining.com/react-router/web/guides/testing.
- I need to pass "initialEntries" to the router so that we can control what the current "path" is, to test routing

Reply:
My thinking is that `application.test.tsx` is a good smoke test to ensure that the whole thing works together using a `BrowserRouter`, kinda like an Integration test, and that this gets into the more nitty gritty tests, like a unit test.

Reply:
After thinking about this, I'd actually like to chat about this a bit more and our decision to use mount() vs shallow() for jest/unit tests - will start a convo in Slack here!